reporter/src/main/kotlin/reporters/AbstractNoticeReporter.kt

```diff
@@ -52,7 +45,22 @@ abstract class AbstractNoticeReporter : Reporter() {
             val headers: List<String>,
             val findings: LicenseFindingsMap,
             val footers: List<String>
-    )
+    ) {
+        fun cleanedFindings(copyrightGarbage: CopyrightGarbage) =
```

Comment:
commit-message: I do not understand from the commit message the following points I'd like to get

1. If garbage is removed in the very first step, why does it become invalid after the postprocessing step is run?
Does the postprocessing re-insert new garbage? What would be an example problem with the ordering prior to this change?

2. The commit message mentions also the copyrightStatementsProcessor, but it does not touch the order of that processing step and the postprocessor script.

3. Is there any reason you would want to insert the garbage into the post processor script? 
If there is no, why not get rid as soon as possible from the garbage an keep on doing it as the very first step.
E.g. if there should be need for debugging, then chances are better you re not messing around with invalid stuff?!
...If the postprocessing script really inserts new garbage, (if yes what would be an example) then there could be an additional garbage removal at the end, before CopyrightStatementsProcessor is called.

Reply:
1. "The post processor script can completely replace the findings..." -> if findings are replaced, the garbage cleanup before is meaningless.
2. Yes it does, before the `CopyrightStatementsProcessor` was called before the post processing script, now it's after.
3. The post processor might want to operate on the raw unprocessed data. Overall the post processor script feature is more or less experimental, so I wouldn't draw conclusions yet about the generic use case. But we know that we currently need it the other way round.

Reply:
Thanks, it becomes clearer to me now. I still think it'd make sense to run garbage removal at the very beginning in addition to where it is run now - because in the use cases we now have the garbage is not needed right from the beginning as far as I understand and if that changes in the future we could re-align if needed. Anyhow, you can keep as is if you prefer